metabench.cmake

```diff
@@ -172,78 +192,115 @@ endfunction()
-"                                                                            \n"
-"stdout                                                                      \n"
-"#{'-'*80}                                                                   \n"
-"#{stdout}                                                                   \n"
-"                                                                            \n"
-"stderr                                                                      \n"
-"#{'-'*80}                                                                   \n"
-"#{stderr}                                                                   \n"
-"                                                                            \n"
-"code                                                                        \n"
-"#{'-'*80}                                                                   \n"
-"#{code}                                                                     \n"
+"require 'benchmark'                                                                                    \n"
+"require 'open3'                                                                                        \n"
+"require 'pathname'                                                                                     \n"
+"require 'ruby-progressbar'                                                                             \n"
+"require 'tempfile'                                                                                     \n"
+"require 'tilt/erb'                                                                                     \n"
+"                                                                                                       \n"
+"                                                                                                       \n"
+"module Metabench                                                                                       \n"
+"  # This function is meant to be called inside a ERB-based `chart.json` file.                          \n"
+"  # `erb_template` should be the name of the `.cpp.erb` file containing the                            \n"
+"  # benchmark to run.                                                                                  \n"
+"  def self.measure(erb_template, range, env: {})                                                       \n"
+"    measure_file = Pathname.new('${METABENCH_DIR}/\@path_to\@/\@target\@.cpp')                         \n"
+"    exe_file = Pathname.new('${METABENCH_DIR}/\@path_to\@/\@target\@${CMAKE_EXECUTABLE_SUFFIX}')       \n"
+"    command = ['${CMAKE_COMMAND}', '--build', '${CMAKE_BINARY_DIR}', '--target', '\@target\@']         \n"
+"    range = range.to_a                                                                                 \n"
+"    range = [range[0], range[-1]] if ENV['METABENCH_TEST_ONLY'] && range.length >= 2                   \n"
```

Comment:
Basically, you'd have to `gem install metabench` and that would install both `metabench.rb` and any dependencies.